storage: TestReplicaRangefeedErrors failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/8bd6caec6eed07ecff29801f3f11993352544387

Parameters:
```
TAGS=
GOFLAGS=
```

Failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=846013&tab=buildLog

```
=== RUN   TestReplicaRangefeedErrors
--- FAIL: TestReplicaRangefeedErrors (1.49s)
=== RUN   TestReplicaRangefeedErrors/REASON_RANGE_MERGED
I180818 06:20:39.160646 124513 storage/store.go:1509  [s1] [n1,s1]: failed initial metrics computation: [n1,s1]: system config not yet available
I180818 06:20:39.164728 124513 gossip/gossip.go:383  [n1] NodeDescriptor set to node_id:1 address:<network_field:"tcp" address_field:"127.0.0.1:41753" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180818 06:20:39.168492 124513 storage/store.go:1509  [s2] [n2,s2]: failed initial metrics computation: [n2,s2]: system config not yet available
W180818 06:20:39.169374 124513 gossip/gossip.go:1351  [n2] no incoming or outgoing connections
I180818 06:20:39.169444 124513 gossip/gossip.go:383  [n2] NodeDescriptor set to node_id:2 address:<network_field:"tcp" address_field:"127.0.0.1:37277" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180818 06:20:39.170244 124592 gossip/client.go:129  [n2] started gossip client to 127.0.0.1:41753
I180818 06:20:39.223119 124513 storage/store.go:1509  [s3] [n3,s3]: failed initial metrics computation: [n3,s3]: system config not yet available
W180818 06:20:39.226984 124513 gossip/gossip.go:1351  [n3] no incoming or outgoing connections
I180818 06:20:39.227049 124513 gossip/gossip.go:383  [n3] NodeDescriptor set to node_id:3 address:<network_field:"tcp" address_field:"127.0.0.1:40117" > attrs:<> locality:<> ServerVersion:<major_val:0 minor_val:0 patch:0 unstable:0 > build_tag:"" started_at:0 
I180818 06:20:39.227796 124938 gossip/client.go:129  [n3] started gossip client to 127.0.0.1:41753
I180818 06:20:39.281186 124513 storage/replica_raftstorage.go:529  [s1,r1/1:/M{in-ax}] generated preemptive snapshot e6fa32df at index 16
I180818 06:20:39.281227 124513 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n2 established
I180818 06:20:39.282225 124513 storage/store_snapshot.go:655  [s1,r1/1:/M{in-ax}] streamed snapshot to (n2,s2):?: kv pairs: 47, log entries: 6, rate-limit: 2.0 MiB/sec, 1ms
I180818 06:20:39.282571 124834 storage/replica_raftstorage.go:773  [s2,r1/?:{-}] applying preemptive snapshot at index 16 (id=e6fa32df, encoded size=8295, 1 rocksdb batches, 6 log entries)
I180818 06:20:39.283672 124834 storage/replica_raftstorage.go:779  [s2,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180818 06:20:39.284057 124513 storage/replica_command.go:792  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n2,s2):2): read existing descriptor r1:/M{in-ax} [(n1,s1):1, next=2, gen=0]
I180818 06:20:39.285733 124513 storage/replica.go:3749  [txn=ee4ee5e0,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n2,s2):2): updated=[(n1,s1):1 (n2,s2):2] next=3
I180818 06:20:39.286437 124513 storage/replica_raftstorage.go:529  [s1,r1/1:/M{in-ax}] generated preemptive snapshot 950ca9ae at index 18
I180818 06:20:39.286476 124513 rpc/nodedialer/nodedialer.go:92  [s1,r1/1:/M{in-ax}] connection to n3 established
I180818 06:20:39.287578 124513 storage/store_snapshot.go:655  [s1,r1/1:/M{in-ax}] streamed snapshot to (n3,s3):?: kv pairs: 50, log entries: 8, rate-limit: 2.0 MiB/sec, 1ms
I180818 06:20:39.288030 124971 storage/replica_raftstorage.go:773  [s3,r1/?:{-}] applying preemptive snapshot at index 18 (id=950ca9ae, encoded size=9374, 1 rocksdb batches, 8 log entries)
I180818 06:20:39.289241 124971 storage/replica_raftstorage.go:779  [s3,r1/?:/M{in-ax}] applied preemptive snapshot in 1ms [clear=0ms batch=0ms entries=0ms commit=1ms]
I180818 06:20:39.289663 124513 storage/replica_command.go:792  [s1,r1/1:/M{in-ax}] change replicas (ADD_REPLICA (n3,s3):3): read existing descriptor r1:/M{in-ax} [(n1,s1):1, (n2,s2):2, next=3, gen=0]
I180818 06:20:39.291092 124739 rpc/nodedialer/nodedialer.go:92  connection to n1 established
I180818 06:20:39.292835 124513 storage/replica.go:3749  [txn=3ffe4c65,s1,r1/1:/M{in-ax}] proposing ADD_REPLICA((n3,s3):3): updated=[(n1,s1):1 (n2,s2):2 (n3,s3):3] next=4
I180818 06:20:39.445585 124513 storage/replica_command.go:282  [s1,r1/1:/M{in-ax}] initiating a split of this range at key "m" [r2]
I180818 06:20:39.451620 124513 storage/replica_command.go:410  [s1,r1/1:{/Min-m}] initiating a merge of r2:{m-/Max} [(n1,s1):1, (n2,s2):2, (n3,s3):3, next=4, gen=0] into this range
I180818 06:20:39.466231 124812 util/stop/stopper.go:537  quiescing; tasks left:
1      storage.intentResolver: processing intents
1      [async] wait-for-merge
I180818 06:20:39.466893 124812 util/stop/stopper.go:537  quiescing; tasks left:
1      storage.intentResolver: processing intents
W180818 06:20:39.470797 124810 storage/intent_resolver.go:738  [s1] failed to cleanup transaction intents: failed to resolve intents: result is ambiguous (server shutdown)
W180818 06:20:39.474076 124983 storage/raft_transport.go:281  unable to accept Raft message from (n3,s3):3: no handler registered for (n1,s1):1
W180818 06:20:39.474259 124981 storage/store.go:3713  [s3,r2/3:{m-/Max}] raft error: node 1 claims to not contain store 1 for replica (n1,s1):1: store 1 was not found
W180818 06:20:39.474276 124979 storage/raft_transport.go:583  while processing outgoing Raft queue to node 1: store 1 was not found:
W180818 06:20:39.475320 124992 storage/raft_transport.go:578  creating batch client for node 1 failed: rpc error: code = Canceled desc = grpc: the client connection is closing
W180818 06:20:39.475568 124985 storage/raft_transport.go:583  while processing outgoing Raft queue to node 3: rpc error: code = Unavailable desc = transport is closing:
    --- FAIL: TestReplicaRangefeedErrors/REASON_RANGE_MERGED (0.34s)
    	replica_rangefeed_test.go:407: merge saw unexpected error: storage/replica_command.go:554: merge of range into 1 failed: waiting for all right-hand replicas to catch up: rpc error: code = Unknown desc = r2 was not found

```

--- a/pkg/storage/replica_rangefeed_test.go
+++ b/pkg/storage/replica_rangefeed_test.go
@@ -363,6 +363,14 @@ func TestReplicaRangefeedErrors(t *testing.T) {
 		}
 		rightRangeID := mtc.Store(0).LookupReplica(splitKey).RangeID
 
+		// Write to the RHS of the split and wait for all replicas to process it.
+		// This ensures that all replicas have seen the split before we move on.
+		incArgs := incrementArgs(roachpb.Key("n"), 9)
+		if _, pErr := client.SendWrapped(ctx, mtc.distSenders[0], incArgs); pErr != nil {
+			t.Fatal(pErr)
+		}
+		mtc.waitForValues(roachpb.Key("n"), []int64{9, 9, 9})
+
 		// Establish a rangefeed on the left replica.
 		streamLeft := newTestStream()
 		streamLeftErrC := make(chan *roachpb.Error, 1)

